ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -197,10 +197,12 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
                                         if (isArchivedWorkflow(workflow) && (globalDeleteArchived || !isWorkflowInCluster(workflow))) {
                                             services.workflows.deleteArchived(workflow.metadata.uid, workflow.metadata.namespace).catch(setError);
```

Comment:
I'm pretty sure that this line and the equivalent above are asynchronous promises. If they're not waited on to finish, the redirect and reload may happen _before_ the request completes.

It is possible the requests get cancelled as a result of that, would need to see some traces to confirm that though (as it depends).

That _could_ be the root cause here, but not really sure without looking in more detail

Reply:
Fixed in #11711